webapp/channels/src/components/suggestion/at_mention_provider/at_mention_suggestion.tsx

```diff
@@ -33,169 +31,167 @@ interface Group extends Item {
-                                }}
-                            />
-                        }
+        icon = (
+            <span className='status-wrapper style--none'>
+                <span className='profile-icon'>
+                    <Avatar
+                        username={item && item.username}
+                        size='sm'
+                        url={Utils.imageURLForUser(item.id, item.last_picture_update)}
                     />
                 </span>
-            );
-        }
+                <StatusIcon status={item && item.status}/>
+            </span>
+        );
 
-        return (
-            <div
-                className={classNames('suggestion-list__item', {'suggestion--selected': isSelection})}
-                data-testid={`mentionSuggestion_${itemname}`}
-                onClick={this.handleClick}
-                onMouseMove={this.handleMouseMove}
-                {...Suggestion.baseProps}
-            >
-                {icon}
-                <span className='suggestion-list__ellipsis'>
-                    <span className='suggestion-list__main'>
-                        {'@' + itemname}
-                    </span>
-                    {item.is_bot && <BotTag/>}
-                    {description}
-                    {youElement}
-                    {customStatus}
-                    {sharedIcon}
-                    {isGuest(item.roles) && <GuestTag/>}
-                </span>
-                {countBadge}
-            </div>
+        customStatus = (
+            <CustomStatusEmoji
+                showTooltip={true}
+                userID={item.id}
+                emojiSize={15}
+                emojiStyle={{
+                    margin: '0 4px 4px',
+                }}
+            />
         );
     }
-}
 
-export default injectIntl(AtMentionSuggestion);
+    const youElement = item.isCurrentUser ? (
+        <FormattedMessage
+            id='suggestion.user.isCurrent'
+            defaultMessage='(you)'
+        />
+    ) : null;
+
+    const sharedIcon = item.remote_id ? (
+        <SharedUserIndicator
+            className='shared-user-icon'
+            withTooltip={true}
+        />
+    ) : null;
+
+    let countBadge;
+    if (item.type === Constants.MENTION_GROUPS) {
+        countBadge = (
+            <span className='suggestion-list__group-count'>
+                <Tag
+                    text={
+                        <FormattedMessage
+                            id='suggestion.group.members'
+                            defaultMessage='{member_count} {member_count, plural, one {member} other {members}}'
+                            values={{
+                                member_count: (item as Group).member_count,
+                            }}
+                        />
+                    }
+                />
+            </span>
+        );
+    }
+
+    return (
+        <SuggestionContainer
```

Comment:
Previously, the `Suggestion` base class provided `handleClick` and `handleMouseMove` methods which was the main reason I'd used inheritance originally. Now, that's encapsulated inside of this `SuggestionContainer` component. The new component also handles setting the class name as well since I realized we were repeating the same class names in every single `*Suggestion` component as well